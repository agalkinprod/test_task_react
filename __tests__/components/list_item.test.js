import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import jest from 'jest-mock';

import ListItem from "../../src/components/list_item/list_item";

describe('<ListItem />', () => {
  describe('render()', () => {

    it('should render the component', () => {
      const wrapper = mount(<ListItem
        id={1}
        text={'test'}
        editTextAction={jest.fn()}
        deleteAction={jest.fn()}
        is_selected={false}
        selectAction={jest.fn()}
      />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should contain a checkbox', () => {
      const wrapper = mount(<ListItem
        id={1}
        text={'test'}
        editTextAction={jest.fn()}
        deleteAction={jest.fn()}
        is_selected={false}
        selectAction={jest.fn()}
      />);
      expect(wrapper.find('.ttr-checkbox-circle_unchecked').length).toEqual(1);
    });

    it('should contain control buttons', () => {
      const wrapper = mount(<ListItem
        id={1}
        text={'test'}
        editTextAction={jest.fn()}
        deleteAction={jest.fn()}
        is_selected={false}
        selectAction={jest.fn()}
      />);
      expect(wrapper.find('.ttr-list_item-action').length).toEqual(2);
    });

    it('should react to select checkbox click', () => {
      const mockOnSelect = jest.fn();
      const wrapper = mount(<ListItem
        id={1}
        text={'test'}
        editTextAction={jest.fn()}
        deleteAction={jest.fn()}
        is_selected={false}
        selectAction={mockOnSelect}
      />);
      wrapper.find('.ttr-checkbox-circle_unchecked').simulate('click');
      expect(mockOnSelect.mock.calls.length).toEqual(1);
    });

    it('should react to edit button click', () => {
      const mockOnEdit = jest.fn();
      const wrapper = mount(<ListItem
        id={1}
        text={'test'}
        editTextAction={mockOnEdit}
        deleteAction={jest.fn()}
        is_selected={false}
        selectAction={jest.fn()}
      />);
      wrapper.find('.ttr-list_item-action').at(0).simulate('click');
      expect(mockOnEdit.mock.calls.length).toEqual(1);
    });

    it('should react to delete button click', () => {
      const mockOnDelete = jest.fn();
      const wrapper = mount(<ListItem
        id={1}
        text={'test'}
        editTextAction={jest.fn()}
        deleteAction={mockOnDelete}
        is_selected={false}
        selectAction={jest.fn()}
      />);
      wrapper.find('.ttr-list_item-action').at(1).simulate('click');
      expect(mockOnDelete.mock.calls.length).toEqual(1);
    });

  });
});