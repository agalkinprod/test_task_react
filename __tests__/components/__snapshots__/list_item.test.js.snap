// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> render() should render the component 1`] = `
<ListItem
  deleteAction={[MockFunction]}
  editTextAction={[MockFunction]}
  id={1}
  is_selected={false}
  selectAction={[MockFunction]}
  text="test"
>
  <div
    className="ttr-list_item"
  >
    <Checkbox
      onClick={[Function]}
      state={false}
    >
      <div
        className="ttr-checkbox"
      >
         
        <div
          className="ttr-checkbox-circle_unchecked"
          onClick={[Function]}
        />
         
      </div>
    </Checkbox>
    <div
      className="ttr-list_item-text"
    >
       
      test
       
    </div>
    <div
      className="ttr-list_item-action"
      onClick={[Function]}
    >
       
      <img
        src="https://icongr.am/fontawesome/pencil.svg?size=22&color=000000"
      />
       
    </div>
    <div
      className="ttr-list_item-action"
      onClick={[Function]}
    >
       
      <img
        src="https://icongr.am/fontawesome/trash.svg?size=22&color=000000"
      />
       
    </div>
  </div>
</ListItem>
`;
